infobind/cache.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infobind
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultBindCacheSize = 5
+
+// bindData store the basic bind info and bindSql astNode.
+type bindData struct {
+	bindRecord
+	ast ast.StmtNode
```

Comment:
This is not used, can we remove this.

Reply:
Can we rename bindData to bindRecord and rename bindRecord to bindMeta?

Reply:
We can't remove this, because it will be used in query sql's bind match.

Reply:
@iamzhoug37 Could you add a comment about the usage of `ast`?

Reply:
Why not directly storing the `ast` field to the `bindMeta` struct?

Reply:
> @iamzhoug37 Could you add a comment about the usage of `ast`?

ok, the ast is using for sql bind replace old sql ast with new ast.



> Why not directly storing the `ast` field to the `bindMeta` struct?

the bindMeta is bind sql table's data.

Reply:
> the bindMeta is bind sql table's data.

but it doesn't matter?

Reply:
It will be used in query sql bind.